Compilation in kernel 6.1 is broken

There is something in 6.1 that has changed the way some of the defined guards for VXWORK work. We get this error:

```
  CC [M]  /home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.o
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c:270:5: error: "_WRS_VXWORKS_MAJOR" is not defined, evaluates to 0 [-Werror=undef]
  270 | #if _WRS_VXWORKS_MAJOR == 7
      |     ^~~~~~~~~~~~~~~~~~
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c: In function 'InitPci':
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c:409:5: error: "_WRS_VXWORKS_MAJOR" is not defined, evaluates to 0 [-Werror=undef]
  409 | #if _WRS_VXWORKS_MAJOR == 7
      |     ^~~~~~~~~~~~~~~~~~
/home/docker/mdis_test_system/actions-runner/_work/13MD05-90/13MD05-90/mdis_test_system/MDIS_Compilation_Test/OBJ/dbg/men_chameleon/chameleonv2.c:515:5: error: "_WRS_VXWORKS_MAJOR" is not defined, evaluates to 0 [-Werror=undef]
  515 | #if _WRS_VXWORKS_MAJOR == 7
      |     ^~~~~~~~~~~~~~~~~~
```

We need to fix this and also find why this only affects 6.1 and not older versions.

--- a/chameleonv2.c
+++ b/chameleonv2.c
@@ -267,7 +267,7 @@ typedef struct {
 /*--------------------------------------*/
 /*	PROTOTYPES     	                    */
 /*--------------------------------------*/
-#if _WRS_VXWORKS_MAJOR == 7
+#if defined(VXWORKS) && (_WRS_VXWORKS_MAJOR == 7)
 IMPORT VXB_DEV_ID sysGetFpgaPciCtrlID(void);
 #endif
 /*--------------------------------------*/
@@ -406,7 +406,7 @@ _STATIC_ int32 InitPci(
 	int32				error=0;
 	void				*tblAddr=NULL;
 	u_int32				barVal;
-#if _WRS_VXWORKS_MAJOR == 7
+#if defined(VXWORKS) && (_WRS_VXWORKS_MAJOR == 7)
 	VXB_DEV_ID pciId = sysGetFpgaPciCtrlID();
 	VXB_RESOURCE * pRes = NULL;
 	VXB_RESOURCE_ADR * pResAdr = NULL;
@@ -512,7 +512,7 @@ _STATIC_ int32 InitPci(
 				h->ba[i].type = -1;	/* unused */
 
 			}
-#if _WRS_VXWORKS_MAJOR == 7
+#if defined(VXWORKS) && (_WRS_VXWORKS_MAJOR == 7)
 			pRes = vxbResourceAlloc (pciId, VXB_RES_MEMORY, (UINT16)i);
 			if (pRes != NULL)
 			{

